ti/config/config.go

```diff
@@ -127,3 +130,16 @@ func (c *Cfg) GetFeatureState(stepID string, feature types.SavingsFeature) (type
 	}
 	return types.DISABLED, ErrStateNotFound
 }
+
+func (c *Cfg) LockZip() {
+	c.zipmu.Lock()
+	atomic.StoreInt32(&c.ziplocked, 1)
```

Comment:
If you are using atomic operations then you dont need then mutex

Reply:
modified logic with atomic operations itself